File: LayoutController/src/main/java/com/aliucord/plugins/layoutcontroller/patchers/MembersInviteButtonPatch.java
package com.aliucord.plugins.layoutcontroller.patchers;

import com.aliucord.plugins.layoutcontroller.patchers.base.BasePatcher;
import com.aliucord.plugins.layoutcontroller.util.Const;
import com.aliucord.plugins.layoutcontroller.util.Util;
import com.discord.databinding.WidgetChannelMembersListItemAddBinding;
import com.discord.widgets.channels.memberlist.adapter.ChannelMembersListViewHolderAdd;

import kotlin.jvm.functions.Function0;
import top.canyie.pine.Pine;

public class MembersInviteButtonPatch extends BasePatcher {

    public MembersInviteButtonPatch() throws Exception {
        super(Const.Key.INVITE_BUTTON_MEMBERS_KEY, Const.Description.INVITE_BUTTON_MEMBERS_DESCRIPTION, ChannelMembersListViewHolderAdd.class.getDeclaredMethod("bind", Function0.class, int.class));
    }

    @Override
    public void patchBody(Pine.CallFrame callFrame) throws Exception {
        var _this = callFrame.thisObject;
        var _binding = _this.getClass().getDeclaredField("binding");
        _binding.setAccessible(true);

        var binding = (WidgetChannelMembersListItemAddBinding) _binding.get(_this);

        if (binding == null) return;

        Util.hideViewCompletely(binding.a);

        callFrame.setResult(callFrame.getResult());
    }
}

File: plugins/LayoutController/src/main/java/com/aliucord/plugins/layoutcontroller/patchers/MembersInviteButtonPatch.kt
package com.aliucord.plugins.layoutcontroller.patchers

import com.aliucord.plugins.layoutcontroller.patchers.base.BasePatcher
import com.aliucord.plugins.layoutcontroller.util.Description
import com.aliucord.plugins.layoutcontroller.util.Key
import com.aliucord.plugins.layoutcontroller.util.hideCompletely
import com.discord.databinding.WidgetChannelMembersListItemAddOrLeaveBinding
import com.discord.widgets.channels.memberlist.adapter.ChannelMembersListViewHolderAdd
import de.robv.android.xposed.XC_MethodHook
import top.canyie.pine.Pine.CallFrame

class MembersInviteButtonPatch : BasePatcher(
    key = Key.INVITE_BUTTON_MEMBERS_KEY,
    description = Description.INVITE_BUTTON_MEMBERS_DESCRIPTION,
    classMember = ChannelMembersListViewHolderAdd::class.java.getDeclaredMethod(
        "bind",
        Function0::class.java,
        Int::class.javaPrimitiveType
    )
) {

    override fun patchBody(callFrame: XC_MethodHook.MethodHookParam) {
        val thisObject = callFrame.thisObject

        val binding = thisObject.javaClass
            .getDeclaredField("binding")
            .let {
                it.isAccessible = true
                it.get(thisObject) as WidgetChannelMembersListItemAddOrLeaveBinding
            }

        binding.a.hideCompletely()
        callFrame.result = callFrame.result
    }

}